hopsworks-api/src/main/java/io/hops/hopsworks/api/featurestore/json/datavalidation/ConstraintDTO.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * This file is part of Hopsworks
+ * Copyright (C) 2019, Logical Clocks AB. All rights reserved
+ *
+ * Hopsworks is free software: you can redistribute it and/or modify it under the terms of
+ * the GNU Affero General Public License as published by the Free Software Foundation,
+ * either version 3 of the License, or (at your option) any later version.
+ *
+ * Hopsworks is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+ * PURPOSE.  See the GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License along with this program.
+ * If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package io.hops.hopsworks.api.featurestore.json.datavalidation;
```

Comment:
If the DTOs are moved to common, are Constraint.java and ContraintGroups.java required?

Reply:
Yes and no. Semantically `Constraint` and `ConstraintGroups` are used only by the controllers and the API transforms them to whatever representation is needed and vice-versa, in this case `ConstraintDTO` and `ConstraintGroupsDTO`. Also, `ConstraintGroupsDTO` can contain multiple Constraints itself - it acts as a container.